gossipd/gossip.c

```diff
@@ -1426,7 +1426,7 @@ static void gossip_refresh_network(struct daemon *daemon)
 static struct io_plan *connection_in(struct io_conn *conn, struct daemon *daemon)
 {
 	struct wireaddr addr;
-	struct sockaddr_storage s;
+	struct sockaddr_storage s = {};
 	socklen_t len = sizeof(s);
 
 	if (getpeername(io_conn_fd(conn), (struct sockaddr *)&s, &len) != 0) {
```

Comment:
Does not either `getpeername` initialize `s`, or we return immediately with a close of the socket?

Reply:
But `getpeername(…)` doesn't necessarily fill the entire `struct sockaddr_storage` (note that `s` is a `struct sockaddr_storage`). That's why the static analyzer is complaining, right? :-)